mozilla/blurts-server
# MNTOR-3248: Stablize HIBP webhook

# References:

Jira: MNTOR-3487, MNTOR-3248



# Description
Resolving some issues related to scaling our postgres database.
We started noticing a lot of connection timeouts for Postgres when we get huge spikes of traffic coming from HIBP when big breaches occur. We (SRE and we) have implemented some workaround via PG Bouncer, but the issue still persist

In this PR, I'm attempting to address it via caching some frequently-used and rarely-changing big objects in Redis.
The `all breaches` object from HIBP is an obvious choice for now and it will be set in Redis the expiration is set to be `1 hour`. 
The appropriate fallback logic follows the below priority:
1. Redis
2. Postgres
3. HIBP API

There will additional PRs to set the object in Redis for the cronjob that fetches the `all breaches` object for Postgres

# Screenshot (if applicable)

Not applicable.

# How to test
* Without Redis, the code will automatically use a mock version of Redis (all the tests, or FE eng, can use this version)
* With Redis installed, set the `REDIS_URL` to something like `redis://localhost:your-port-of-choice` and you should be able to see the breaches object populated in Redis with the key `breaches`

# Checklist (Definition of Done)
